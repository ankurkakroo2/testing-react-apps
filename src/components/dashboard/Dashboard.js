import React, { useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import { AccountCircle, Menu as MenuIcon } from "@material-ui/icons";

import {
  AppBar,
  Menu,
  MenuItem,
  IconButton,
  Toolbar,
  Typography
} from "@material-ui/core";

import { getUser, removeUserSession } from "../../util/Common";
import PlaceHolder from "./Placeholder";
import "./default.css";

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  }
}));

const Dashboard = props => {
  const classes = useStyles();
  const user = getUser();
  const [anchorEl, setAnchorEl] = useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  // handle click event of logout button
  const handleLogout = () => {
    removeUserSession();
    props.history.push("/login");
  };

  return (
    <section className="dashboard">
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Dashboard
          </Typography>
          <div>
            <IconButton
              id="account-icon-button"
              aria-label="account of current user"
              aria-controls="menu-appbar"
              aria-haspopup="true"
              onClick={handleClick}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>
            <Menu
              id="account-menu"
              anchorEl={anchorEl}
              keepMounted
              open={Boolean(anchorEl)}
              onClose={handleClose}
            >
              <MenuItem id="menu-item-username" onClick={() => {}}>
                Signed in as: {user}
              </MenuItem>
              <MenuItem id="menu-item-profile" onClick={() => {}}>
                Profile
              </MenuItem>
              <MenuItem id="menu-item-logout" onClick={handleLogout}>
                Logout
              </MenuItem>
            </Menu>
          </div>
        </Toolbar>
      </AppBar>
      <PlaceHolder count={6}/>
    </section>
  );
};

export default Dashboard;
