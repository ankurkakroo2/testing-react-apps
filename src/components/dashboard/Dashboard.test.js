import React from "react";
import Dashboard from "./Dashboard";
import { render } from "@testing-library/react";

let result;
beforeEach(() => {
  result = render(<Dashboard />);
});

describe("Dashboard", () => {
  test("render without crash", () => {
    expect(result.getByText("Welcome", { exact: false })).toBeInTheDocument();
  });

  test("snapshot", () => {
    expect(result.asFragment()).toMatchSnapshot();
  });
});
