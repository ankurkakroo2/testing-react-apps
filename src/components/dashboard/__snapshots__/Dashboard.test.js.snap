// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapshot 1`] = `
<DocumentFragment>
  <section
    class="dashboard"
  >
    <div
      class="dashboard__header"
    >
      <div
        class="dashboard__header__profile-section"
      >
        <div
          class="dashboard__header__profile-icon"
        />
        <input
          class="dashboard__logout-btn"
          type="button"
          value="Logout"
        />
      </div>
    </div>
    <p>
      Welcome !
    </p>
    <div
      class="box-placeholder"
    >
      <div
        class="p-4"
      >
        <span
          class="category text link"
        />
        <h4
          class="text line"
        />
        <h4
          class="text"
        />
      </div>
      <hr />
      <div
        class="image"
      >
        <div
          class="embed-responsive embed-responsive-16by9"
        />
      </div>
      <hr />
      <div
        class="excerpt p-4"
      >
        <div
          class="text line"
        />
        <div
          class="text line"
        />
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="box-placeholder"
    >
      <div
        class="p-4"
      >
        <span
          class="category text link"
        />
        <h4
          class="text line"
        />
        <h4
          class="text"
        />
      </div>
      <hr />
      <div
        class="image"
      >
        <div
          class="embed-responsive embed-responsive-16by9"
        />
      </div>
      <hr />
      <div
        class="excerpt p-4"
      >
        <div
          class="text line"
        />
        <div
          class="text line"
        />
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="box-placeholder"
    >
      <div
        class="p-4"
      >
        <span
          class="category text link"
        />
        <h4
          class="text line"
        />
        <h4
          class="text"
        />
      </div>
      <hr />
      <div
        class="image"
      >
        <div
          class="embed-responsive embed-responsive-16by9"
        />
      </div>
      <hr />
      <div
        class="excerpt p-4"
      >
        <div
          class="text line"
        />
        <div
          class="text line"
        />
        <div
          class="text"
        />
      </div>
    </div>
    <div
      class="box-placeholder"
    >
      <div
        class="p-4"
      >
        <span
          class="category text link"
        />
        <h4
          class="text line"
        />
        <h4
          class="text"
        />
      </div>
      <hr />
      <div
        class="image"
      >
        <div
          class="embed-responsive embed-responsive-16by9"
        />
      </div>
      <hr />
      <div
        class="excerpt p-4"
      >
        <div
          class="text line"
        />
        <div
          class="text line"
        />
        <div
          class="text"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
